### ISSUE TITLE
Documentation Updates

### ISSUE THREAD

**Issue by tpill90** [tpill90/steam-lancache-prefill#216]
- [ ] Multi-account support
	- [x] Use steam family sharing to share a secondary account with the primary account used to prefill
	- [ ] Use two separate instances of the app (two folders, two containers, etc), that each login to their own account
- [ ] Document password security
	- [x] #278
	- [x] 2FA is the ideal way to go if worried about account security
- [x] FAQ
	- [x] Login timeout errors, and manifest download failures can be transient.  If they continue to happen after waiting awhile (6 hours - 1 day), then there may be something wrong with either SteamPrefill or your setup.
	- [x] Mention that SteamPrefill doesn't need to be ran on your server.  It may be easier or more convenient to use your windows desktop
	- [x] Add a section for "Can I prefill games I don't own?"

**Comment by tmontney** [comment]
Additionally, how long do login sessions last? From time to time, I'll manually run prefill and it'll prompt me to sign in again. I almost never have to sign-in again with the official Steam client.

**Comment by cybersteel8** [comment]
I would like to request an enhancement to the documentation provided here as well
https://tpill90.github.io/steam-lancache-prefill/Detailed-Command-Usage/

> Running with the flag --force will override this behavior, and instead will always run the prefill, re-downloading all files **for the specified product**

This phrase is mentioned in the documentation for the --force flag, but there is no documentation stating how to specify a product. For example, I would like to force a prefill of a certain product that is already in my select-apps list, so I would like to use the --force flag in addition to specifying the product I would like to force (instead of the entire select-apps list) but documentation on how to do this is not provided.

**Comment by tpill90** [comment]
Hi @cybersteel8 . Thanks for bringing this to my attention. 

I realize now that the documentation you're referring to was actually a copy and paste from the documentation for BattlenetPrefill, which requires you to specify the product codes to prefill. 

Currently there is no way to specify that only a subset of apps that should be used with the `--force` flag, it will always use all of the selected apps. 

What I would be interested to hear is what use case you're running into where you'd like to only force one app, but not all.  Is it a matter of you having a large library, or it takes a long time to rerun the prefill on your hardware?  Possibly both?  Or is it something else entirely?

**Comment by cybersteel8** [comment]
Hi @tpill90, thanks for the quick response!
I have a large selection, approximately 40 selected games, and I recently had to clear the entire cache (I did this as a troubleshooting step to a LANCache problem) and I have run a force prefill of all of my select games. I have noticed through it's execution that some games have "1 failed downloads" next to them, but the items did finish.

I would like to specifically force a prefill of those games again to ensure they are correctly cached.

```
[2:28:39 PM] Starting Tiny Tina's Wonderlands
[2:28:51 PM] Downloading 48.33 GiB
1 failed downloads
[3:36:35 PM] Finished in 1:07:43.64 - 102.16 Mbit/s
```
I also have an entry that has this message in the middle of its prefill
`An unexpected error (System.Net.Http.HttpRequestException) occurred while downloading manifests.  Retrying...`
So for sanity, I would like to also select that app to force a prefill.

The prefill of 40 games from an empty cache does take a very long time, well over 12 hours. I could run the force command again and it will take significantly less time since the majority of the entries are now cached, but it is still in the order of hours, due to hardware limitations. It would be quicker to select the individual apps I'd like to force a prefill on rather than the entire list of selected apps.

**Comment by tpill90** [comment]
This all makes sense to me!

So to clarify, Tiny Tina's was previously prefilled and considered as a success, prior to you emptying your lancache. 

You then emptied the cache, and had to rerun every selected app with the --force flag, because otherwise SteamPrefill would consider them "up to date".

When prefilling with --force, one of the requests failed, and is listed as failing to correctly prefill. However SteamPrefill still considers it as "up to date/successful", because of the prefill you did prior to clearing the cache?

Is that all correct?

**Comment by cybersteel8** [comment]
>So to clarify, Tiny Tina's was previously prefilled and considered as a success, prior to you emptying your lancache.

Correct.

>You then emptied the cache, and had to rerun every selected app with the --force flag, because otherwise SteamPrefill would consider them "up to date".

Exactly right. SteamPrefill considered my cache "up to date" despite it being empty.

>When prefilling with --force, one of the requests failed, and is listed as failing to correctly prefill. However SteamPrefill still considers it as "up to date/successful", because of the prefill you did prior to clearing the cache?

This part I cannot verify. The force prefill (from the previous step) is still running. I don't know how to verify that it "is listed as failing to correctly prefill" and I cannot yet verify that "SteamPrefill considers it up to date/successful" as I am yet to run a normal (without the --force flag) prefill.

This is an opportunity for you to suggest what I should test for your benefit. I am still at the stage where I am force prefilling every selected app, I expect there is about a hour or so to go.

**Comment by tpill90** [comment]
Once your --force run is complete, I would like to know if Tiny Tina is still marked as failing.  And I would like to know if you run prefill without the --force flag, if Tiny Tina is still marked as "up to date"

I have a feeling that the real issue here is SteamPrefill's awareness of the cache being cleared, or it's handling of failed apps when running with --force.  Some potential solutions off the top of my head:

1) If an app fails when running with `--force`, then SteamPrefill will consider that app as "not up to date".  It will then only consider it "up to date" again if any subsequent prefills are successful. This would solve your issue of having to specify a subset of apps to prefill, as the next prefill without `--force` would only be for that one failed app. However I think this might be awkward and unituitive from a user's perspective, as well as not being the intended purpose of `--force`

2) Perhaps there could be an additional command added that could reset the tracked status of successfully prefilled apps,  essentially resetting back to your initial prefill run.  In this case prefill would work exactly as you would have expected, allowing you to retry only the failed apps.  I can see this command being primarily intended for this exact use case, where you have cleared your cache and need to make SteamPrefill aware of it.

**Comment by cybersteel8** [comment]
```
[5:24:53 PM] Prefill complete!
──────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────
  Prefilled 50 apps in 15:24:31.64

   Updated │ Up To Date
  ━━━━━━━━━┿━━━━━━━━━━━━
     48    │     0

[5:24:53 PM] Disconnected from Steam!
```

The forced prefill finally completed. I noticed immediately that it says 50 apps were prefilled, but only 48 are updated. I believe the other app might be Euro Truck Simulator 2 that also shows "1 failed downloads".
The game Teardown had the unexpected error I mentioned in an earlier comment:
`An unexpected error (System.Net.Http.HttpRequestException) occurred while downloading manifests.  Retrying...`

Anyway, I will now run a normal prefill without the force flag.

```
# ./SteamPrefill prefill
[5:33:31 PM] Starting login!
[5:33:33 PM] Logged into Steam
[5:33:33 PM] Steam session initialization complete!

[5:33:36 PM] Starting Phasmophobia
[5:33:39 PM] Downloading 8 GiB
[5:33:39 PM] Detected Lancache server at lancache.steamcontent.com [<IP REDACTED>]
[5:37:24 PM] Finished in 03:45.47 - 304.82 Mbit/s

[5:37:24 PM] Prefill complete!
──────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────
  Prefilled 50 apps in 03:53.83

   Updated │ Up To Date
  ━━━━━━━━━┿━━━━━━━━━━━━
      1    │     49

[5:37:24 PM] Disconnected from Steam!
```

This tells me that SteamPrefill considers all games up-to-date. Phasmophobia had no error attached to it when I ran the force prefill, so I assume there was an update to the game that SteamPrefill detected. It appears that Tiny Tina, ETS2 and Teardown are all considered up-to-date.

**Comment by tpill90** [comment]
@cybersteel8 Would you mind breaking out all of this info from our discussion into its own issue?  There is certainly something that needs to be fixed here, I'm just not certain exactly what yet.  And I don't want to forget about all of this.

**Comment by tpill90** [comment]
> Additionally, how long do login sessions last? From time to time, I'll manually run prefill and it'll prompt me to sign in again. I almost never have to sign-in again with the official Steam client.

With the most recent release, it should now be a 6 month auth token.
